specification/maps/data-plane/Weather/preview/1.0/weather.json

```diff
@@ -593,19 +1017,52 @@
     "HazardIndex": {
       "type": "integer",
       "format": "int32",
-      "description": "A severity/hazard index.\n  * `0` - No hazard.\n  * `1` - Be informed, be aware.\n  * `2` - Pay attention, be prepared.\n  * `3` - Take action.\n  * `4` - Life threatening, emergency."
+      "description": "A severity/hazard index.\n  * `0` - No hazard.\n  * `1` - Be informed, be aware.\n  * `2` - Pay attention, be prepared.\n  * `3` - Take action.\n  * `4` - Life threatening, emergency.",
+      "enum": [
+        0,
+        1,
+        2,
+        3,
+        4
+      ],
+      "x-ms-enum": {
+        "name": "HazardIndex",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": 0,
+            "description": "No hazard."
+          },
+          {
+            "value": 1,
+            "description": "Be informed, be aware."
+          },
+          {
+            "value": 2,
+            "description": "Pay attention, be prepared."
+          },
+          {
+            "value": 3,
+            "description": "Take action."
+          },
+          {
+            "value": 4,
+            "description": "Life threatening, emergency."
+          }
+        ]
+      }
     },
     "HourlyForecast": {
       "type": "object",
       "properties": {
         "date": {
+          "x-ms-client-name": "dateTime",
           "type": "string",
           "description": "Date and time of the forecast in ISO 8601 format, for example, 2019-10-27T19:39:57-08:00.",
           "format": "date-time"
         },
         "iconCode": {
-          "type": "integer",
-          "format": "int32",
+          "$ref": "#/definitions/IconCode",
           "description": "Numeric value representing an image that displays the `iconPhrase`. Please refer to [Weather Service Concepts](https://aka.ms/AzureMapsWeatherConcepts) for details."
```

Comment:
This `description` can be deleted since it is also in the `IconCode` definition.